app/controllers/proposal/collection_draft_proposals_controller.rb

```diff
@@ -95,13 +99,48 @@ def progress
 
       @fourth_information = get_progress_message('done') if get_resource.proposal_status == 'done'
 
-      @available_actions =  if get_resource.in_work?
-                              'Make additional changes or submit this proposal for approval.'
-                            elsif get_resource.submitted? || get_resource.rejected?
-                              'You may rescind this proposal to make additional changes.'
-                            else
-                              'No actions are possible.'
-                            end
+      @available_actions =  get_available_actions_text
+    end
+
+    def approve
+      authorize get_resource
+
+      add_status_history('approved')
+
+      if get_resource.approve && get_resource.save
+        Rails.logger.info("Audit Log: User #{current_user.urs_uid} approved #{resource_name.titleize} with title: '#{get_resource.entry_title}' and id: #{get_resource.id}")
+
+        user_from_resource_response = user_from_resource
+        # User e-mail
+        ProposalMailer.proposal_approved_notification(user_from_resource_response, get_resource.draft['ShortName'], get_resource.draft['Version'], params['id']).deliver_now if user_from_resource_response
+        # Approver e-mail
+        ProposalMailer.proposal_approved_notification(get_user_info, get_resource.draft['ShortName'], get_resource.draft['Version'], params['id']).deliver_now
```

Comment:
Is this only going to the approver that approved the proposal? I may be mistaken, but I thought they wanted this email to go to all the approvers

Reply:
Do they want each approver to get an e-mail each time any of them approves things?  Seems spammy to me.  I will check with Alicia, but I think that particular request was for submissions.

Reply:
Hmm, I think you are right, that was for submissions.